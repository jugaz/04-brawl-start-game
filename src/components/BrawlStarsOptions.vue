<template>
   <div class="options-container">
      <ul>
         <li class="name-brawl" v-for="brawl in brawls" :key="brawl.id" @click="$emit('selectionBrawlStart', brawl.id)">
            {{ brawl.name }}
         </li>
      </ul>
   </div>
</template>

<script>
   export default {
      props: {

         brawls: {
            type: Array,
            required: true
         },
      },
   }
</script>

<style scoped>
   ul {
      list-style-type: none;
   }

   li {
      background-color: white;
      border-radius: 5px;
      color: #fcd34d;
      border: 1px solid #fcd34d;
      cursor: pointer;
      margin-bottom: 10px;
      padding: 5px 0;
      width: 250px;
   }

   li:hover {
      box-shadow: 0 3px 2px 2px #00000033;
   }

   .options-container {
      display: flex;
      justify-content: center;
   }

   li.success {
      border: 1px solid #2bb673;
      color: #2bb673;
      animation-iteration-count: infinite;
      animation-name: pulse;
      animation-duration: 1s;
      animation-delay: .1s;
   }

   li.fail {
      border: 1px solid red;
      color: red;
      animation-name: shakeX;
      animation-duration: 1s;
      animation-delay: .1s;
   }
   li.fail,
   li.success,
   li.disabled {
      pointer-events: none;
      cursor: none;
   }


</style>