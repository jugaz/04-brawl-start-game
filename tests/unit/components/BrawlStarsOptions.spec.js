import { shallowMount } from "@vue/test-utils";
import BrawlStarsOptions from "@/components/BrawlStarsOptions";

import { mockBrawlstars } from "../mocks/brawlstarts.mock";

describe("BrawlStarsOptions component", () => {

    let wrapper

    beforeEach( () => {
        wrapper = shallowMount( BrawlStarsOptions, {
            props: {
                brawls: mockBrawlstars
            }
        } )
    })

    test('debe de hacer match con el snapshot',() => {
        
        expect( wrapper.html() ).toMatchSnapshot()
    
    })
    
    test('debe de mostrar las 4 opciones correctamente',() => {

        const liTags = wrapper.findAll("li")
        expect( liTags.length ).toBe(4)

        expect(liTags[0].text()).toBe('SHELLY')
        expect(liTags[1].text()).toBe('COLT')
        expect(liTags[2].text()).toBe('BULL')
        expect(liTags[3].text()).toBe('BROCK')
    
    })

    test('debe de emitir "selectionBrawlStart" con sus respectivos parÃ¡metros al hacer click',() => {
        
        const [ li1, li2, li3, li4 ] = wrapper.findAll("li")

        li1.trigger("click")
        li2.trigger("click")
        li3.trigger("click")
        li4.trigger("click")

        expect( wrapper.emitted('selectionBrawlStart').length ).toBe(4)
        expect( wrapper.emitted('selectionBrawlStart')[0] ).toStrictEqual( [1] )
        expect( wrapper.emitted('selectionBrawlStart')[1] ).toStrictEqual( [2] )
        expect( wrapper.emitted('selectionBrawlStart')[2] ).toStrictEqual( [3] )
        expect( wrapper.emitted('selectionBrawlStart')[3] ).toStrictEqual( [4] )

    
    })
    
})