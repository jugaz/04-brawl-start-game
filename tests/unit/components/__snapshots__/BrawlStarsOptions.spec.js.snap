// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrawlStarsOptions component debe de hacer match con el snapshot 1`] = `
<div class="options-container">
  <ul>
    <li class="name-brawl">SHELLY</li>
    <li class="name-brawl">COLT</li>
    <li class="name-brawl">BULL</li>
    <li class="name-brawl">BROCK</li>
  </ul>
</div>
`;
